### PR TITLE
Add new "GameMode" option which implements org.freedesktop.portal.GameMode

### PR DESCRIPTION
**Pull request by Noted-Jrelvas**
Currently, users have to manually edit configuration to use gamemode by setting a custom launcher. This is bad because the majority of distros have gamemode present. Furthermore, our flatpak requires an extension to be installed separately for `gamemoderun` to be available, which further degrades user experience.

This PR fixes that by implementing support for requesting gamemode to be enabled via desktop portals, which is done by default (for both player and studio). This should work out of the box for most distros and for flatpak installs.

Note: VinegarBridge will depend on the new `internal/bus` package for some of its functionality, so this PR is a prerequisite for implementing it.

### CODE REVIEW

**Comment by Noted-Jrelvas** [comment]
Systems without dbus and/or desktop portals will simply not use this new functionality. There should be no regressions.

File: internal/bus/gamemode.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		return false, call.Err
+	}
+
+	response := call.Body[0].(int32)
+
+	if response < 0 {
+		return false, nil
+	}
+
+	return true, nil
```
Comment (apprehensions) [inline internal/bus/gamemode.go:33]:
```suggestion
	return response > 0, nil
```

File: internal/bus/bus.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	}
+
+	portal := conn.Object("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop")
+
+	session := &SessionBus{
+		conn: conn,
+		portal: portal,
+	}
+
+	return session
```
Comment (apprehensions) [inline internal/bus/bus.go:25]:
```suggestion
	session := &SessionBus{
		conn: conn,
		portal: conn.Object("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop"),
	}
```

File: config/config.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ type Binary struct {
 	FFlags        roblox.FFlags `toml:"fflags"`
 	Env           Environment   `toml:"env"`
 	ForcedGpu     string        `toml:"gpu"`
+	GameMode      bool        `toml:"gamemode"`
```
Comment (apprehensions) [inline config/config.go:38]: `go fmt`

**Review by apprehensions — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 3 above)
Comment (Noted-Jrelvas) [reply config/config.go]: Fixed by 0bebc3d6f4f72cec8b792f5c693fd05de886cf41

**Comment by Noted-Jrelvas** [comment]
I've applied your feedback and suggestions. I'm ready for a second review.

File: internal/bus/bus.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	call := s.portal.Call("org.freedesktop.portal.GameMode.RegisterGame", 0, int32(pid))
+	if call.Err != nil {
+		//Transparently handle missing portal
+		if !errors.Is(call.Err, dbus.ErrMsgNoObject) {
+			return nil
+		}
+		return call.Err
+	}
+
+	return nil
```
Comment (apprehensions) [inline internal/bus/bus.go:40]:
```suggestion
	call := s.portal.Call("org.freedesktop.portal.GameMode.RegisterGame", 0, pid)
	if call.Err != nil && !errors.Is(call.Err, dbus.ErrMsgNoObject) {
		return call.Err
	}

	return nil
```

File: internal/bus/bus.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		return &SessionBus{}
+	}
+
+	return &SessionBus{
+		conn:   conn,
+		portal: conn.Object("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop"),
+	}
+}
+
+func (s *SessionBus) GamemodeRegister(pid int) (error) {
```
Comment (apprehensions) [inline internal/bus/bus.go:26]:
```suggestion
func (s *SessionBus) GamemodeRegister(pid int32) (error) {
```

**Review by apprehensions — CHANGES_REQUESTED** [review]

File: internal/bus/bus.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		return &SessionBus{}
+	}
+
+	return &SessionBus{
+		conn:   conn,
+		portal: conn.Object("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop"),
+	}
+}
+
+func (s *SessionBus) GamemodeRegister(pid int) error {
```
Comment (Noted-Jrelvas) [inline internal/bus/bus.go:26]:
```suggestion
func (s *SessionBus) GamemodeRegister(pid int32) error {
```

File: internal/bus/bus.go (hunk 5 above)
Comment (Noted-Jrelvas) [reply internal/bus/bus.go]: Fixed.

**Review by apprehensions — APPROVED** [review]

**Review by lunarlettuce — APPROVED** [review]

**Comment by RackYican** [comment]
It is actived by default or we need to activate it?

**Comment by Noted-Jrelvas** [comment]
It's active by default

**Pull request merged**